sdk/eventhub/azure-eventhubs/azure/eventprocessorhost/eh_partition_pump.py

```diff
@@ -45,18 +45,37 @@ def __init__(self, host, lease):
             await self.processor.process_error_async(self.partition_context, last_exception)
             self.set_pump_status("OpenFailed")
 
+        _started_ok = False
+        _retry_count = 0
+
         if self.pump_status == "Opening":
             loop = asyncio.get_event_loop()
-            self.set_pump_status("Running")
-            await self.eh_client.run_async()
-            self.running = loop.create_task(self.partition_receiver.run())
+            while (not _started_ok) and (_retry_count < 5):
+                try:
+                    await self.eh_client.run_async()
+                    _started_ok = True
+                except Exception as err:
+                    _logger.warning(
+                        "%r,%r PartitionPumpWarning: Failure starting client or receiver, retrying: %r",
+                        self.host.guid, self.partition_context.partition_id, err)
+                    last_exception = err
+                    _retry_count += 1
+
+            if not _started_ok:
+                _logger.warning(
+                    "%r,%r PartitionPumpWarning: Failure starting client or receiver: %r",
+                    self.host.guid, self.partition_context.partition_id, last_exception)
+                await self.process_error_async(self.partition_context, last_exception)
```

Comment:
While this should technically work - I would define `last_exception` initially up on line 50 just for clarity (unless it's already defined elsewhere that I didn't see?)

Reply:
Sounds reasonable, just added